components/grid/selection/rows.md

```diff
@@ -0,0 +1,282 @@
+
+## SelectedItemsChanged
+
+You can respond to the user action of selecting a new row through the `SelectedItemsChanged` event. The `SelectedItemsChanged` event receives a collection of the Grid data model. It may have no items in it. It may have only one member (the last selected item) when the `SelectionMode` is `Single`.
+
+>caption Single selection with one-way binding for SelectedItems and using the SelectedItemsChanged event
+
+````CSHTML
+<TelerikGrid Data=@GridData
+             SelectionMode="@GridSelectionMode.Single"
+             SelectedItemsChanged="@((IEnumerable<Employee> employeeList) => OnSelect(employeeList))"
+             SelectedItems="@SelectedEmployees"
+             Pageable="true"
+             Height="400px">
+    <GridColumns>
+        <GridCheckboxColumn />
+        <GridColumn Field=@nameof(Employee.Name) />
+        <GridColumn Field=@nameof(Employee.Team) Title="Team" />
+    </GridColumns>
+</TelerikGrid>
+
+@if (SelectedEmployees != null)
+{
+    <ul>
+        @foreach (Employee employee in SelectedEmployees)
+        {
+            <li>
+                @employee.Name
+            </li>
+        }
+    </ul>
+}
+
+@code {
+    private List<Employee> GridData { get; set; }
+    private IEnumerable<Employee> SelectedEmployees { get; set; } = Enumerable.Empty<Employee>();
+    private Employee SelectedEmployee { get; set; }
+    protected override void OnInitialized()
+    {
+        GridData = new List<Employee>();
+        for (int i = 0; i < 15; i++)
+        {
+            GridData.Add(new Employee()
+                {
+                    EmployeeId = i,
+                    Name = "Employee " + i.ToString(),
+                    Team = "Team " + i % 3
+                });
+        }
+    }
+
+    protected void OnSelect(IEnumerable<Employee> employees)
+    {
+        SelectedEmployee = employees.FirstOrDefault();
+        // update the collection so that the grid can highlight the correct item
+        // when two-way binding is used this happens automatically,
+        // but the framework does not allow two-way binding and the event at the same time
+        SelectedEmployees = new List<Employee> { SelectedEmployee };
+    }
+
+    public class Employee
+    {
+        public int EmployeeId { get; set; }
+        public string Name { get; set; }
+        public string Team { get; set; }
+    }
+}
+````
+
+### SelectedItemsChanged and Asynchronous Operations
+
+If you want to execute asynchronous operations, such as loading data on demand, when the user selects a row, this should be handled in the [`OnRowClick`]({%slug grid-events%}#onrowclick) or [`OnRowDoubleClick`]({%slug grid-events%}#onrowdoubleclick) events rather than in the [`SelectedItemsChanged`]({%slug grid-events%}#selecteditemschanged).
```

Comment:
```suggestion
To execute asynchronous operations, such as loading data on demand, when the user selects a row, handle these operations in the [`OnRowClick`]({%slug grid-events%}#onrowclick) or [`OnRowDoubleClick`]({%slug grid-events%}#onrowdoubleclick) events rather than in the [`SelectedItemsChanged`]({%slug grid-events%}#selecteditemschanged).
```